dev/devicelab/lib/tasks/perf_tests.dart

```diff
@@ -1459,8 +1434,49 @@ class CompileTest {
         throw Exception('Unsupported option for fake devices');
       case DeviceOperatingSystem.fuchsia:
         throw Exception('Unsupported option for Fuchsia devices');
+      case DeviceOperatingSystem.ios:
       case DeviceOperatingSystem.macos:
-        throw Exception('Unsupported option for macOS devices');
+        unawaited(stderr.flush());
+        late final String deviceId;
+        if(deviceOperatingSystem == DeviceOperatingSystem.ios) {
+          deviceId = 'ios';
+        } else if (deviceOperatingSystem == DeviceOperatingSystem.macos) {
+          deviceId = 'macos';
+        } else {
+          throw Exception('Attempted to run darwin compile workflow with $deviceOperatingSystem');
+        }
+
+        options.insert(0, deviceId);
+        options.add('--tree-shake-icons');
+        options.add('--split-debug-info=infos/');
+        watch.start();
+        await flutter('build', options: options);
+        watch.stop();
+        final Directory buildDirectory = dir(path.join(
+          cwd,
+          'build',
+        ));
+        final String? appBundlePath =
+            _findDarwinAppInBuildDirectory(buildDirectory.path);
+        if (appBundlePath == null) {
+          throw 'Failed to find app bundle in ${buildDirectory.path}';
+        }
+        final Directory appBundle = Directory(appBundlePath);
+        releaseSizeInBytes = await appBundle
```

Comment:
By no longer compressing, this will increase the iOS size benchmark trend.  The engine sheriff may contact you about this.

@zanderso

Reply:
Please retain the compression step for iOS.

Reply:
@zanderso Is this metric used for something I'm not aware of?

Reply:
We track release size in SkiaPerf https://flutter-flutter-perf.skia.org/e/?queries=sub_result%3Drelease_size_bytes%26test%3Dhello_world_ios__compile

Reply:
Oh for sure, I've looked at that data. Is it a problem to rebenchmark it to make it consistent between macOS and iOS? The more we can make the darwin tests and benchmarks the same between iOS and macOS, the less maintenance burden it is since improvements to one platform make it to the other.  It also makes it more meaningful to compare benchmarks across platforms.  We could compress the macOS app but that doesn't really mean anything.  We could also make an iOS specific code path to compress it, but I don't care so much whether we're measuring it compressed or not, as long as we're keep tracking of it growing or shrinking over time.

Reply:
Yes, we do want to track the compressed size. An example of a reason to track both compressed and uncompressed size is that a change to the Dart snapshot format might make the uncompressed size smaller while making the compressed size bigger due to compressing less well as a result of a formatting or data layout change.

Reply:
Does it make sense to do the same for macOS then?

Reply:
@zanderso thanks for letting us know. 

This is the result from running `_compileApp` on `iOS` and `macOS` for this patch. I think it's what we need. 

https://github.com/flutter/flutter/blob/5c1b21c4f97190a9b3144cda029cc626ffa03ee8/dev/devicelab/lib/tasks/perf_tests.dart#L1475-L1478

https://github.com/flutter/flutter/blob/5c1b21c4f97190a9b3144cda029cc626ffa03ee8/dev/devicelab/lib/tasks/perf_tests.dart#L1572-L1575

Reply:
I think if we're tracking compressed size for one target, then that is sufficient, but it certainly doesn't hurt to collect it for more platforms if that simplifies this code.